File: FLAME_qwen/data/compiler_technology/buggy/1/1337/wrong/parser/expression/unaryexp/Character.cpp
/**
 * Character:
 */

#include <fstream>
#include "Character.h"
#include "../../../token/Token.h"
#include "../../../util/Util.h"

void Character::characterParser() {
    tokenItem token = Token::getNowToken();
    if(token.type == "CHRCON"){
        token = Token::getNextToken();
    }
    Util::printParserInfo("<Character>");
}

File: FLAME_project/data/compiler_technology/buggy/1/1350/wrong/nonTerminal/Character.java
package nonTerminal;

import frontend.Lexer;
import frontend.Token;

import java.util.ArrayList;

public class Character {
    private Token chrcon;
    private int lineCnt;

    public Character() {
        this.chrcon = null;
        this.lineCnt = 0;
    }

    public void parser(Lexer lexer) {
        chrcon = lexer.now();
        lexer.move();
        lineCnt = chrcon.getLineCnt();
    }

    public int getLineCnt() {
        return lineCnt;
    }

    public void analyze(ArrayList<String> symbolTable, ArrayList<String> errors,
                        ArrayList<Integer> domainNums, int domainCnt) {

    }
}
